src/hooks/reviewHooks/useMergeReviewRequest.ts

```diff
@@ -0,0 +1,43 @@
+import { AxiosError } from "axios"
+import {
+  UseMutationResult,
+  useMutation,
+  useQueryClient,
+  QueryClient,
+} from "react-query"
+
+import {
+  REVIEW_REQUEST_QUERY_KEY,
+  SITE_DASHBOARD_REVIEW_REQUEST_KEY,
+} from "constants/queryKeys"
+
+import { ErrorDto } from "types/error"
+
+import * as ReviewService from "../../services/ReviewService"
+
+export const useMergeReviewRequest = (
+  siteName: string,
+  prNumber: number,
+  shouldInvalidate = true
```

Comment:
yeap, it's just used there. i exposed this so that if we do use it somewhere else in the future, we can get invalidation by default.

Reply:
if we do remove the invalidation prop, we haev to decide whether we want to manually invalidate or invalidate in the hook.

invalidation in the hook would lead to a desync with the desired flow (on publish, we will trigger a refetch -> go back to dashboard rather than showing the modal). manual desync leads to breaking of established code convention + usages in the future have to be aware of this (burden of knowledge).

i felt that this was a compromise between the 2 but lmk what are your thoughts!

Reply:
Oh yeah I'm ok with having validation/invalidation but it was more of whether there would be a chance that we will use this hook again somewhere else. It's a very small point so we can just leave it as it is!